### PR TITLE
Added a pin icon

### PR DESCRIPTION
**Pull request by Velin92**
Note: The preview tests require a delay, and the delay function to set them in the tests seems to be partially broken, it only applies the delay on iPad but not on iPhone tests.

Simulator Screenshot - iPhone 15 - 2024-09-10 at 18 47 02

### CODE REVIEW

File: ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift [hunk 1]
```diff
@@ -146,6 +153,7 @@ struct TimelineItemBubbledStylerView<Content: View>: View {
                     context.send(viewAction: .handleTimelineItemMenuAction(itemID: timelineItem.id, action: action))
                 }
             }
+            .pinnedIndicator(isPinned: isPinned, isOutgoing: timelineItem.isOutgoing)
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift:156]: Feels very wasteful to wrap every single item in an HStack when the chances of it being pinned are slim to none. Can we rewrite this to bail out early please?

File: ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift [hunk 2]
```diff
@@ -555,4 +613,97 @@ struct TimelineItemBubbledStylerView_Previews: PreviewProvider, TestablePreview
         }
         .environmentObject(viewModel.context)
     }
+        
+    // These akwats include a reply
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift:617]:
```suggestion
    // These **dont' know what it's actually supposed to be** include a reply
```

**Review by stefanceriu — CHANGES_REQUESTED** [review]

**Comment by stefanceriu** [comment]
>  The preview tests require a delay, and the delay function to set them in the tests seems to be partially broken, it only applies the delay on iPad but not on iPhone tests.

Not entirely sure what's wrong there but all the old tests were passing when the new snapshotting flows got merged. Let's tackle whatever it is separately from this PR

File: ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift (hunk 1 above)
Comment (Velin92) [reply ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift:156]: Sure

File: ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift (hunk 2 above)
Comment (Velin92) [reply ElementX/Sources/Screens/Timeline/View/Style/TimelineItemBubbledStylerView.swift:617]: just some copy and paste garbage, removed

**Review by stefanceriu — APPROVED** [review]
Nice, ship it! 🚀

**Pull request merged**